### DISCUSSION TITLE
How about a plugin for private pages?

### DISCUSSION THREAD

**Discussion by kanbeq** [datenstrom/community#388] [Ask a question] [answered]
For luck =), I need a plugin that will allow you to block access to the page and unlock it via a password. I am asking for help in creating or at least considering my proposal. If this plugin is available, I will not have to look for another CMS anymore. This would looks like [this](https://github.com/Diyzzuf/grav-plugin-private-site).

**Comment by markseuffert** [comment]
[How about the draft plugin?](https://github.com/annaesvensson/yellow-draft) You have to log in with your user account before you can read draft pages.

Hope this helps. 😄

**Comment by kanbeq** [comment]
I would not like to give outsiders the password to the admin panel, let
alone about to give outsiders the opportunity to edit the hidden content. I
mean, the access just had a group of people who knows the password.

24.12.2018 12:08 PM "Mark Seuffert" <notifications@github.com> napisał(a):

**Comment by markseuffert** [comment]
You can restrict user accounts, see https://github.com/annaesvensson/yellow-edit

**Comment by kanbeq** [comment]
Could you give me an example of what to put in this file?

24.12.2018 12:54 PM "Mark Seuffert" <notifications@github.com> napisał(a):

**Comment by schulle4u** [comment]
The problem is that users who are restricted to their home directory can still edit pages inside this directory, which is not suitable in each case. I think what we need here is a very basic password plugin to individually protect pages. It doesn't necessarily have to use the same security level as the user accounts. Perhaps something like my [spoiler plugin](https://github.com/schulle4u/yellow-spoiler), but with a better content hiding mechanism. The password could be stored inside a page setting, and the page content is fetched if the user enters the right password into a simple form. I'm however not sure yet how this can be done the simplest way. 

Have a nice holiday. 
Steffen

**Comment by kanbeq** [comment]
Yes! That's exactly what I mean. If you could find a way to create such a
plugin, I would be very grateful. Thanks for your wishes and also wish you
happy holidays.

24.12.2018 1:32 PM "Steffen Schultz" <notifications@github.com> napisał(a):

**Comment by schulle4u** [comment]
@bialy03 During a rather boring christmas dinner with my family I found a really simple solution using Yellow's templating capabilities. Please have a look here: 
https://robbenradio.de/test/about/

If you enter `test123` into the password box and hit login, you should see a dummy text. Is this what you are looking for? This needs to be made a little more bulletprof, but it should solve the problem. I can release this as a plugin during the next days. 

FYI: The solution consists of a snippet called `content-private.php, and a `private` template file. To protect a page, your page settings can look like this: 

```
---
Title: Protected page
Password: secret
Template: private
---
Here is something very secret.
```

Steffen

**Comment by kanbeq** [comment]
Yes. This is exactly what I wanted. It will be the perfect Christmas
present for me.

pon., 24 gru 2018 o 23:40 Steffen Schultz <notifications@github.com>
napisał(a):

**Comment by schulle4u** [comment, marked as answer]
Well then, let's see what santa has left for you. :) 

https://github.com/schulle4u/yellow-private/

As mentioned above, you have to add password and template settings to your protected pages. Additionally you should add a description setting as well as a page break at the beginning of your text, this will remove page content from site metadata and page listings (e.g. feed plugin). More information in the README. Please test and report any issues, I'll make this an official release if everything works. 

Steffen

**Reply by binyominzeev** [reply]
Now it's a 404, the new link is here: https://github.com/schulle4u/yellow-private

**Reply by schulle4u** [reply]
Thanks for pointing out. I've updated the repository link in the original post.

**Comment by kanbeq** [comment]
Everything works. I do not know how to thank you;). The only thing that would come in handy is some information that an incorrect password has been entered (of course, if this event takes place).

**Comment by schulle4u** [comment]
Currently there are no existing text strings in Yellow I could use for this, even the edit plugin doesn't have a wrong password message. I of course can create language files for the plugin and add some more convenience to the template, however I can only provide german and english for now.

**Comment by kanbeq** [comment]
Ok. I understand. I can translate into Polish. Nevertheless, I still thank
you for making the plugin.

25.12.2018 5:03 PM "Steffen Schultz" <notifications@github.com> napisał(a):

**Comment by kanbeq** [comment]
![image](https://user-images.githubusercontent.com/37421974/50491200-724ecf00-0a11-11e9-914b-40f23871eb2d.png)
Can not somehow "blur"?
You can not find the password in the link you gave earlier, and you can not see this strange script.

**Comment by schulle4u** [comment]
Can you please describe the image and the problem? I'm visually impaired and therefore cannot get anything useful out of it. A quick OCR prints a file not found error, but I don't understand the problem yet. :)

**Comment by kanbeq** [comment]
The point is that for those wanting access, nothing difficult. The password
can be read from the page code. So it is not very safe. And here the
question: Is it possible to hide it somehow?

27.12.2018 8:46 PM "Steffen Schultz" <notifications@github.com> napisał(a):

**Comment by schulle4u** [comment]
OK, I think I understand now. If you are logged in, the edit plugin loads all relevant site data into a JSON array, including non-public data. As Yellow doesn't use any database backend, this information is required for plugin operation, but never will be included in publically viewable pages, including ones with private plugin active. Please log out and re-open the source code of your protected page, then you'll only see what's intended for the public.

**Comment by kanbeq** [comment]
It actually is. It looks like the plugin has no errors and you can issue it
officially. You do not even need to translate it because words "password"
and "log in" are already translated from the general translations.

27.12.2018 10:06 PM "Steffen Schultz" <notifications@github.com> napisał(a):
